Zend/zend_compile.c

```diff
@@ -5206,9 +5226,14 @@ void zend_compile_func_decl(znode *result, zend_ast *ast) /* {{{ */
 	}
 
 	zend_compile_params(params_ast, return_type_ast);
-	if (uses_ast) {
-		zend_compile_closure_uses(uses_ast);
+	if (op_array->fn_flags & ZEND_ACC_CLOSURE) {
```

Comment:
yeah, it's a leftover of my first version(which use static_variables[0] for this_ptr, so there is extra logic for all closures), I will remove it.